src/eos/eosEngine.ts

```diff
@@ -1018,6 +1167,9 @@ export class EosEngine extends CurrencyEngine<EosTools> {
   async broadcastTx(
     edgeTransaction: EdgeTransaction
   ): Promise<EdgeTransaction> {
+    if (!this.getResourcesMutex) {
+      await this.getResources()
```

Comment:
The way you have written `getResources`, it should be safe to call without any checks. So, it would be safer to remove the `if` statement here, so we don't leak that implementation detail out of the function.